lib/ansible/plugins/callback/default.py

```diff
@@ -213,7 +215,8 @@ def _print_task_banner(self, task):
         if task_name is None:
             task_name = task.get_name().strip()
 
-        self._display.banner(u"%s [%s%s]" % (prefix, task_name, args))
+        checkmsg = " [CHECK MODE]" if task.check_mode or (self._play_context and self._play_context.check_mode) else ""
```

Comment:
i would not depend on play_context, the play should reflect the correct check_mode for itself at that point

Reply:
There is an issue that I fixed in a new force commit. It shoud read
```
        checkmsg = " [CHECK MODE]" if task.check_mode or (task.check_mode is None and self._play_context and self._play_context.check_mode) else ""
```

The intended and checked behavior is :
- When the play has a defined check_mode this applies. 
- Else the runtime config from _play_context applies

This is kind of duplication of logic of task options and playconext runtime arguments merging, that is implemented somewhere in deep in TaskExectutor.